### ISSUE TITLE
go-mode should set `adaptive-fill-regexp`

### ISSUE THREAD

**Issue by timcharper** [dominikh/go-mode.el#119]
Presently, `adaptive-fill-regexp` is not set, causing `fill-region` to not reformat comments properly.

I believe the regexp from `c-mode` can be safely borrowed (since commenting `//` and `/*` are the same). But there's probably some special comment syntax in one lang or the other of which I'm not aware.

Anyways, I borrowed the one from `c-mode` and it works fine. I call this in a function included in `go-mode-hook`.

```
(set (make-local-variable 'adaptive-fill-regexp)
       "[   ]*\\(//+\\|\\**\\)[     ]*\\([  ]*\\([-–!|#%;>*·•‣⁃◦]+[  ]*\\)*\\)")
```

**Comment by dominikh** [comment]
Is this related to #91? Also, is there a solution that doesn't require a quite as horrifying regular expression? Maybe it can be stripped down? At a glance, it looks like it handles a lot more special formatting than we really need.

**Comment by dominikh** [comment]
Also, when you say `c-mode`, do you mean `cc-mode`? Because in `cc-mode`, I can't find where it'd be setting that variable, and `cc-mode` uses its own function for filling (`c-fill-paragraph`)

**Comment by timcharper** [comment]
Horrifying reg ex is the only option of which in aware. I pulled this one from c-mode

**Comment by codeasone** [comment]
That really is a horrific reg ex but it's working for me. Thanks @timcharper.

**Comment by codeasone** [comment]
Actually, I spoke too soon. It fails for the following 

```
////////////////////////////////////////////////////////////////////////////////
// Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam,
////////////////////////////////////////////////////////////////////////////////
``` 

After `M-x fill-paragraph` with the above selected, I'm getting: 

```
////////////////////////////////////////////////////////////////////////////////
// Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod //
tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam,
////////////////////////////////////////////////////////////////////////////////
``` 

But still, the work-around you suggested @timcharper helps in a number of other scenarios.

**Comment by codeasone** [comment]
Given some more testing, that reg ex is unreliable for my comment reformatting/filling use cases and I wouldn't recommend using it. Shame because this workflow is really well supported in Clojure, Ruby, and Python modes.

**Comment by sdemos** [comment]
It seems like `rust-mode` uses something called `adaptive-fill-function` instead of `adaptive-fill-regex` to the same end - https://github.com/rust-lang/rust-mode/blob/master/rust-mode.el#L1547. They also set a bunch of other things to allow for filling regions in comments. Maybe something like that could minimize or remove the need for "horrifying regexes" and be nicer to maintain.

Being able to reflow golang comments with `fill-region` is something that I've sorely missed.

**Comment by dominikh** [comment]
Thanks, I shall look into that.

**Comment by muirrn** [comment]
Fixed in f04f060eec8f3fba3bd8d02e5c18503d63dfc6a2. Let me know if find issues filling comments.
